internal/caching/set.go

```diff
@@ -66,7 +76,25 @@ func (ss *Set) ContainsOrAdd(v []byte) bool {
 		ss.flop, ss.flip = ss.flip, ss.flop
 		log.Debugw("Cleared flop and swapped subsets as max size is reached", "maxSize", ss.maxSize)
 	}
-	return false
+	return false, nil
+}
+
+func (ss *Set) newKey(namespace, v []byte) ([32]byte, error) {
+	hasher, err := blake2b.New(blake2b.Size256, namespace)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	_, err = hasher.Write(v)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	keyB := hasher.Sum(nil)
+	var key [32]byte
+	copy(key[:], keyB)
```

Comment:
```suggestion
	var key [32]byte
	keyB := hasher.Sum(key[:0])
	copy(key[:], keyB)
```

If we do it this way, we can _in theory_ remove the final copy. But I'd just leave it there.